File: donghwa/0x05/10828.cpp
// endl 을 쓰는 것보다 '\n'을 쓰는것이 더 빠르다. 
#include <bits/stdc++.h>
using namespace std;

int main(void){
    ios::sync_with_stdio(0);
    cin.tie(0);
    int N;
    cin >> N;
    stack<int> S;
    while (N--){
        string cmd;
        cin >> cmd;
        if (cmd=="push"){
            int x; 
            cin >> x;
            S.push(x);
        } else if (cmd=="pop"){
            if (S.empty()) cout << -1 << '\n';
            else {
                cout << S.top() << '\n'; 
                S.pop();
            }
        } else if (cmd=="size"){
            cout << S.size() << '\n';
        } else if (cmd=="empty"){
            if (S.empty()) cout << 1 << '\n';
            else cout << 0 << '\n';
        } else {
            if (S.empty()) cout << -1 << '\n';
            else cout << S.top() << '\n';
        }
    }
}

File: donghwa/0x05/10828.py
import sys

N = int(sys.stdin.readline().rstrip())
cmds = [sys.stdin.readline().rstrip() for _ in range(N)]
stack = []
for cmd in cmds :
    if cmd[:4] == "push" :
        x = cmd.split()[1]
        stack.append(x)
    elif cmd == "pop" :
        if len(stack) > 0:
            print(stack.pop())
        else :
            print(-1)
    elif cmd == "size":
        print(len(stack))
    elif cmd == "empty":
        if len(stack) > 0 :
            print(0)
        else :
            print(1)
    elif cmd == "top":
        if len(stack) > 0 :
            print(stack[-1])
        else :
            print(-1)